usaspending_api/search/delta_models/subaward_search.py

```diff
@@ -0,0 +1,555 @@
+    "product_or_service_description": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_legal_entity_country_code": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_legal_entity_country_name": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_legal_entity_county_code": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_legal_entity_county_name": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_legal_entity_zip5": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_legal_entity_city_code": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_legal_entity_congressional": {"delta": "STRING", "postgres": "TEXT"},
+    "place_of_perform_scope": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_place_of_perform_country_co": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_place_of_perform_country_name": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_place_of_perform_county_code": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_place_of_perform_county_name": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_place_of_perform_zip5": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_place_of_perform_city_code": {"delta": "STRING", "postgres": "TEXT"},
+    "sub_place_of_perform_congressio": {"delta": "STRING", "postgres": "TEXT"},
+}
+SUBAWARD_SEARCH_DELTA_COLUMNS = {k: v["delta"] for k, v in SUBAWARD_SEARCH_COLUMNS.items()}
+SUBAWARD_SEARCH_POSTGRES_COLUMNS = {k: v["postgres"] for k, v in SUBAWARD_SEARCH_COLUMNS.items()}
+
+SUBAWARD_SEARCH_POSTGRES_VECTORS = {
+    "keyword_ts_vector": ['sub_awardee_or_recipient_legal', 'product_or_service_description', 'subaward_description'],
+    "award_ts_vector": ['award_piid_fain', 'subaward_number'],
+    "recipient_name_ts_vector": ['sub_awardee_or_recipient_legal'],
+}
+SUBAWARD_SEARCH_POSTGRES_COLUMNS.update({col: "TSVECTOR" for col in SUBAWARD_SEARCH_POSTGRES_VECTORS})
+
+subaward_search_create_sql_string = rf"""
+    CREATE OR REPLACE TABLE {{DESTINATION_TABLE}} (
+        {", ".join([f'{key} {val}' for key, val in SUBAWARD_SEARCH_DELTA_COLUMNS.items()])}
+    )
+    USING DELTA
+    LOCATION 's3a://{{SPARK_S3_BUCKET}}/{{DELTA_LAKE_S3_PATH}}/{{DESTINATION_DATABASE}}/{{DESTINATION_TABLE}}'
+"""
+
+subaward_search_load_sql_string = fr"""
+    WITH location_summary AS (
+        SELECT
+            -- DISTINCT ON (UPPER(feature_name), state_alpha)
+            UPPER(feature_name) as feature_name,
+            state_alpha,
+            county_numeric,
+            UPPER(county_name) as county_name,
+            census_code,
+            ROW_NUMBER() OVER (PARTITION BY UPPER(feature_name), state_alpha ORDER BY UPPER(feature_name), state_alpha, county_sequence, coalesce(date_edited, date_created) DESC, id DESC) as row_num
+        FROM
+            global_temp.ref_city_county_state_code
+        WHERE
+            feature_class = 'Populated Place'
+            AND COALESCE(feature_name, '') <>  ''
+            AND COALESCE(state_alpha, '') <> ''
+        ORDER BY
```

Comment:
Might be able to remove this `ORDER BY` and help to speed this up some (although a small table so maybe not a whole lot). The order by was most likely in place for the `DISTINCT ON`, but since the `ROW_NUMBER` approach is used here there is probably no need to ORDER BY all of the results.

Reply:
Good call. Removing